From f9703ad1566d37b769f1efb4ee8d67bca117532f Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Sat, 9 May 2020 17:28:29 +0800
Subject: [PATCH 19/24] qwaylandwindow: Support setting window activate

Support setting window activate through app_id, for example:
app_id = "requests=activate"

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/client/qwaylandwindow.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index d1e1e5c82..a83d9df63 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -1039,6 +1039,14 @@ void QWaylandWindow::requestActivateWindow()
 {
     // no-op. Wayland does not have activation protocol,
     // we rely on compositor setting keyboard focus based on window stacking.
+
+    if (!mShellSurface) {
+        qWarning("Wayland does not support QWindow::requestActivate()");
+        return;
+    }
+
+    // HACK: Set window activate through app id
+    mShellSurface->setAppId(QLatin1String("requests=activate"));
 }
 
 void QWaylandWindow::unfocus()
-- 
2.20.1

